seastar/src/ffi_utils.rs

```diff
@@ -15,6 +15,28 @@ where
     fn_once_caller::<Func, Ret>
 }
 
+/// Convert the pointer to Box<Func> and then call it but don't consume.
+fn fn_caller<Func, Ret>(raw_func: *mut u8) -> Ret
+where
+    Func: Fn() -> Ret,
+{
+    unsafe {
+        let boxed_func = Box::from_raw(raw_func as *mut Func);
```

Comment:
I'm not sure if it is legal to temporarily convert to a Box and then release. If a panic happens during `boxed_func()`, then the panic will cause the `boxed_func` to be dropped, which is not what we want.

I think you can just convert `raw_func` to `*mut Func`, then `&Func` and then just call it.

Reply:
Yeah this makes sense.